test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -2752,6 +2746,73 @@ TEST_F(ClientLbInterceptTrailingMetadataTest, BackendMetricDataMerge) {
   EXPECT_EQ(total_num_rpcs, num_trailers_intercepted());
 }
 
+struct ClientLbLoadReportTestParams {
+  OrcaLoadReport reported;
+  OrcaLoadReport received;
+};
+
+class ClientLbLoadReportTest : public ClientLbInterceptTrailingMetadataTest,
+                               public ::testing::WithParamInterface<
+                                   struct ClientLbLoadReportTestParams> {};
+
+TEST_P(ClientLbLoadReportTest, Basic) {
+  const int kNumServers = 1;
+  const int kNumRpcs = 10;
+  StartServers(kNumServers);
+  auto response_generator = BuildResolverResponseGenerator();
+  auto channel =
+      BuildChannel("intercept_trailing_metadata_lb", response_generator);
+  auto stub = BuildStub(channel);
+  response_generator.SetNextResolution(GetServersPorts());
+  for (size_t i = 0; i < kNumRpcs; ++i) {
+    CheckRpcSendOk(DEBUG_LOCATION, stub, false, &GetParam().reported);
+    auto actual = backend_load_report();
+    ASSERT_TRUE(actual.has_value());
+    CheckLoadReportAsExpected(*actual, GetParam().received);
+  }
+  // Check LB policy name for the channel.
+  EXPECT_EQ("intercept_trailing_metadata_lb",
+            channel->GetLoadBalancingPolicyName());
+  EXPECT_EQ(kNumRpcs, num_trailers_intercepted());
+}
+
+INSTANTIATE_TEST_SUITE_P(
```

Comment:
I think using a parameterized test is overkill here.  It makes the code harder to understand, and it generates test names that do not actually help a human understand the purpose of the test.

Instead, please just add a helper method to the existing `ClientLbInterceptTrailingMetadataTest` framework to run a test with specified input and expected output, like this:

```
static void RunPerRpcMetricReportingTest(
    const OrcaLoadReport& reported,
    const OrcaLoadReport& expected) {
  // ...run test...
}
```

Then write a separate `TEST_F()` block for each test case you want to test, with an appropriate test name, which calls the helper function with appropriate values.  For example:

```
TEST_F(ClientLbInterceptTrailingMetadataTest, IgnoresOutOfRangeValues) {
  RunPerRpcMetricReportingTest(...);
}

TEST_F(ClientLbInterceptTrailingMetadataTest, AllowsCpuUtilizationGreaterThan1) {
  RunPerRpcMetricReportingTest(...);
}
```

Reply:
Please refactor as I suggested.  I think the parameterized test interface here is much harder to read, and we simply don't need the complexity in this case.

Reply:
Updated.